arrow/arrow.py

```diff
@@ -66,6 +67,14 @@ class Arrow(object):
     def __init__(
         self, year, month, day, hour=0, minute=0, second=0, microsecond=0, tzinfo=None
     ):
+        if sys.version_info[:2] < (3, 6):  # pragma: no cover
+            with warnings.catch_warnings():
+                warnings.simplefilter("default")
```

Comment:
Do we need these? Don't we just need to emit the warning and let the default warning behavior handle the rest?

Reply:
The way python handles these warnings is kinda counter intuitive.

`default` means that the warning will be shown. DeprecationWarning only shows when the module is run as `__main__` on 3.7+, it won't show on 2.7/3.5/3.6 unless we force it.

Kris is using the `catch_warnings` context manager to make sure we reset the warning filter after we modify it. We should probably make it specific to `DeprecationWarning` though.

`warning.simplefilter("default", DeprecationWarning)`